--- a/ca/spottedleaf/dataconverter/util/CommandArgumentUpgrader.java
+++ b/ca/spottedleaf/dataconverter/util/CommandArgumentUpgrader.java
@@ -130,7 +_,7 @@
         for (final Consumer<CommandDispatcher<CommandSourceStack>> extra : builder.extra) {
             extra.accept(this.dispatcher);
         }
-        ExecuteCommand.register(this.dispatcher, context);
+        ExecuteCommand.register(this.dispatcher, context, this.dispatcher); // Kaboom
         ReturnCommand.register(this.dispatcher);
         // This looks weird, but it's what vanilla does when loading functions for datapacks
         this.source = new CommandSourceStack(
