--- a/src/main/java/io/papermc/paper/command/brigadier/PaperCommands.java
+++ b/src/main/java/io/papermc/paper/command/brigadier/PaperCommands.java
@@ -37,6 +_,7 @@
 
     private @Nullable LifecycleEventOwner currentContext;
     private @MonotonicNonNull CommandDispatcher<CommandSourceStack> dispatcher;
+    public @MonotonicNonNull CommandDispatcher<CommandSourceStack> vanillaDispatcher; // Kaboom
     private @MonotonicNonNull CommandBuildContext buildContext;
     private boolean invalid = false;
 
@@ -53,6 +_,14 @@
                 return commands.getDispatcher();
             }
         });
+        // Kaboom start
+        this.vanillaDispatcher = new CommandDispatcher<>(new ApiMirrorRootNode() {
+            @Override
+            public CommandDispatcher<net.minecraft.commands.CommandSourceStack> getDispatcher() {
+                return commands.vanillaDispatcher;
+            }
+        });
+        // Kaboom end
         this.buildContext = commandBuildContext;
     }
 
